src/aliased_buffer.h

```diff
@@ -141,6 +141,30 @@ class AliasedBuffer {
       return aliased_buffer_->GetValue(index_);
     }
 
+    template <typename T>
+    inline Reference& operator+=(const T& val) {
+      const T current = aliased_buffer_->GetValue(index_);
+      aliased_buffer_->SetValue(index_, current + val);
+      return *this;
+    }
+
+    inline Reference& operator+=(const Reference& val) {
+      this->operator+=(static_cast<NativeT>(val));
+      return *this;
+    }
+
+    template <typename T>
+    inline Reference& operator-=(const T& val) {
+      const T current = aliased_buffer_->GetValue(index_);
+      aliased_buffer_->SetValue(index_, current - val);
```

Comment:
Unless I’m missing something, you could just do `operator+=(index_, -val)` here. If you add an overload for the unary minus operator (or is that even necessary?), that should get rid of the overload below that takes in a `const Reference&`.

Reply:
I'll take a closer look at your suggestion on Monday (little short of time right now). Thanks

Reply:
@TimothyGu Please take a look and see if this was what you had in mind, thanks.

Reply:
@TimothyGu If possible I'd like get this merged but it would be great if you could take a look at the latest change and see if this is what you had in mind. Thanks